### ISSUE TITLE
Fresh installed authentication operator keeps "Degraded" status.

### ISSUE THREAD

**Issue by hksdpc250** [openshift/cluster-authentication-operator#566]
When installing OKD on bare metal, all operators except authentication successfully reach Available status.

```
$ oc get clusteroperators
NAME                                       VERSION                          AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.10.0-0.okd-2022-03-07-131213   True        False         True       14h     CustomRouteControllerDegraded: Ingress.config.openshift.io "cluster" is invalid: [status.componentRoutes.currentHostnames: Invalid value: "oauth-openshift.apps.okd.ict620": status.componentRoutes.currentHostnames in body must be of type hostname: "oauth-openshift.apps.okd.ict620", status.componentRoutes.defaultHostname: Invalid value: "oauth-openshift.apps.okd.ict620": status.componentRoutes.defaultHostname in body must be of type hostname: "oauth-openshift.apps.okd.ict620"]...
baremetal                                  4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
cloud-controller-manager                   4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
cloud-credential                           4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
cluster-autoscaler                         4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
config-operator                            4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
console                                    4.10.0-0.okd-2022-03-07-131213   True        False         False      5h4m
csi-snapshot-controller                    4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
dns                                        4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
etcd                                       4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
image-registry                             4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
ingress                                    4.10.0-0.okd-2022-03-07-131213   True        False         False      5h4m
insights                                   4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
kube-apiserver                             4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
kube-controller-manager                    4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
kube-scheduler                             4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
kube-storage-version-migrator              4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
machine-api                                4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
machine-approver                           4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
machine-config                             4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
marketplace                                4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
monitoring                                 4.10.0-0.okd-2022-03-07-131213   True        False         False      38h
network                                    4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
node-tuning                                4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
openshift-apiserver                        4.10.0-0.okd-2022-03-07-131213   True        False         False      14h
openshift-controller-manager               4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
openshift-samples                          4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
operator-lifecycle-manager                 4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
operator-lifecycle-manager-catalog         4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
operator-lifecycle-manager-packageserver   4.10.0-0.okd-2022-03-07-131213   True        False         False      39h
service-ca                                 4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
storage                                    4.10.0-0.okd-2022-03-07-131213   True        False         False      40h
```

`openshift-install wait-for install-complete` successfully returned and I can login to the web console.

```
$ ./openshift-install --dir ict-okd wait-for install-complete
INFO Waiting up to 40m0s (until 11:55AM) for the cluster at https://api.okd.ict620:6443 to initialize...
INFO Waiting up to 10m0s (until 11:25AM) for the openshift-console route to be created...
INFO Install complete!
INFO To access the cluster as the system:admin user when using 'oc', run 'export KUBECONFIG=/home/stream/openshift/ict-okd/auth/kubeconfig'
INFO Access the OpenShift web-console here: https://console-openshift-console.apps.okd.ict620
INFO Login to the console with user: "******", and password: "******"
INFO Time elapsed: 0s
```

Full MESSAGE for authentication operator:
```
CustomRouteControllerDegraded: Ingress.config.openshift.io "cluster" is invalid: [status.componentRoutes.currentHostnames: Invalid value: "oauth-openshift.apps.okd.ict620": status.componentRoutes.currentHostnames in body must be of type hostname: "oauth-openshift.apps.okd.ict620", status.componentRoutes.defaultHostname: Invalid value: "oauth-openshift.apps.okd.ict620": status.componentRoutes.defaultHostname in body must be of type hostname: "oauth-openshift.apps.okd.ict620"]
OAuthServerRouteEndpointAccessibleControllerDegraded: ingress.config/cluster does not yet have status for the "openshift-authentication/oauth-openshift" route
```

`okd.ict620` is only resolvable in cluster.

I added an OpenID Connect identity provider and it works as expected.

**Comment by stlaz** [comment]
There's a BZ that seems to have fixed the issue in 4.11: https://bugzilla.redhat.com/show_bug.cgi?id=2039256. They seem to intend to backport all the way to 4.8.

**Comment by hksdpc250** [comment]
Install version 4.7 then upgrade to 4.10?
